storage/innobase/log/log0recv.cc

```diff
@@ -4142,6 +4142,32 @@ static
   return end_lsn;
 }
 
+#ifdef XTRABACKUP
+/** recalucate file start_lsn by scanning the first block
+and calculationg the start LSN
+@param[in,out]  log                     redo log
+@param[in,out]  checkpoint_lsn          log sequence number found in checkpoint
+header
+*/
+inline static void set_redo_file_start_lsn(log_t &log, lsn_t start_lsn) {
+  auto file = log.m_files.find(start_lsn);
+  auto file_handle = file->open(Log_file_access_mode::READ_ONLY);
+
+  /* read the first redo block and find out heder number */
+  const dberr_t err = log_data_blocks_read(file_handle, LOG_FILE_HDR_SIZE,
+                                           OS_FILE_LOG_BLOCK_SIZE, log.buf);
+  ut_a(err == DB_SUCCESS);
```

Comment:
This is not good style to suppress "returns". ie by adding release assertions.

I suggest to return dberr_t and may be keep the out parameter as LSN.

or you can return both using std::tuple<>

Reply:
fixed